editor/plugins/sprite_frames_editor_plugin.cpp

```diff
@@ -2337,7 +2337,11 @@ bool SpriteFramesEditorPlugin::handles(Object *p_object) const {
 	if (animated_sprite_3d && *animated_sprite_3d->get_sprite_frames()) {
 		return true;
 	}
-	return !frames_editor->is_editing() && Object::cast_to<SpriteFrames>(p_object);
+	SpriteFrames *frames = Object::cast_to<SpriteFrames>(p_object);
+	if (frames && (frames_editor->get_sprite_frames().is_null() || frames_editor->get_sprite_frames() == frames)) {
```

Comment:
I don't get the last condition. "if the new SpriteFrames is the same as already edited SpriteFrames then edit them again" is basically what this means, unless I'm missing something.
This could mean that the old reference is not cleared properly and/or the editor tries to edit the same resource twice.

Reply:
When renaming an animation, the `frames` variable in `frames_editor` is still there with the current SpriteFrames. So the problem was that the previous `is_edited` method was returning true, causing the plugin to close. Maybe `frames` should have been cleared on a rename?

Reply:
Renaming an animation will update the inspector, which leads to `handles()` + `edit()` and then another `handles()` call. So I guess the original assumption that the editor won't check the same resource twice was wrong. It correctly edits it only once though, so it's fine.